src/components/Layout/TruncateText/TruncateText.js

```diff
@@ -0,0 +1,75 @@
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import { useCallback, useState } from 'react'
+
+const TruncateText = ({
+  text,
+  numberOfLines,
+  pTagClassName,
+  linkClassName,
+}) => {
+  const [textOverflow, setTextOverflow] = useState(false)
+  const [isExpanded, setIsExpanded] = useState(false)
+
+  const onShowMoreClick = () => {
+    setIsExpanded(true)
+  }
+
+  const onShowLessClick = () => {
+    setIsExpanded(false)
+  }
+
+  const pRef = useCallback((node) => {
+    if (node != null) {
+      if (node.scrollHeight > node.clientHeight + 1) {
+        setTextOverflow(true)
+      }
+    }
+  })
+
+  return (
+    <>
+      <p
+        ref={pRef}
+        className={cx(
+          pTagClassName ? pTagClassName : '',
+          isExpanded
+            ? ''
+            : `truncate-description truncate-line-${numberOfLines}`
+        )}
+      >
+        {text}
+      </p>
+      {textOverflow ? (
+        !isExpanded ? (
+          <a
+            className={cx('lbh-link', linkClassName ? linkClassName : '')}
+            href="#"
+            onClick={onShowMoreClick}
+          >
+            show more
+          </a>
+        ) : (
+          <a
+            className={cx('lbh-link', linkClassName ? linkClassName : '')}
+            href="#"
+            onClick={onShowLessClick}
+          >
+            show less
+          </a>
+        )
+      ) : (
+        ''
+      )}
+    </>
+  )
+}
+
+TruncateText.propTypes = {
+  text: PropTypes.string.isRequired,
+  numberOfLines: PropTypes.string.isRequired,
+  pTagClassName: PropTypes.string,
+  linkClassName: PropTypes.string,
+}
+
+export default TruncateText
```

Comment:
It would be great if this is renamed to `src/components/Layout/TruncateText/index.js`, so later when you want to include it you can do this
```
import TruncateText from '../Layout/TruncateText'
```